jobs/UnitTest/PRGateUnitTest.groovy

```diff
@@ -0,0 +1,51 @@
+String stash_manifest_name
+String stash_manifest_path
+String repo_dir
+def unit_test
+
+def setRepoDir(repo_dir){
+    this.repo_dir = repo_dir
+}
+
+def setUnitTest(){
+    this.unit_test = load(repo_dir + "/jobs/UnitTest/UnitTest.groovy")
+}
+
+def setManifest(String manifest_name, String manifest_path){
+    this.stash_manifest_name = manifest_name
+    this.stash_manifest_path = manifest_path
+    unit_test.setManifest(manifest_name, manifest_path)
+}
+
+def setTestRepos(){
+    dir(repo_dir){
+        unstash "$stash_manifest_name"
+    }
+    env.MANIFEST_FILE_PATH = repo_dir + "/${this.stash_manifest_path}"
+    sh '''#!/bin/bash
+    pushd ''' + "$repo_dir" + '''
+    ./build-release-tools/HWIMO-BUILD ./build-release-tools/application/parse_manifest.py \
+    --manifest-file $MANIFEST_FILE_PATH \
+    --parameters-file downstream_file
+    '''
+    def repos_need_unit_test = ""
+    if(fileExists ('downstream_file')) {
```

Comment:
can you please explain the purpose and magic of downstream_file ? (comment in code)